pkg/apis/stable/v1alpha1/gameserver.go

```diff
@@ -330,16 +332,23 @@ func (gs *GameServer) GetDevAddress() (string, bool) {
 	return devAddress, hasDevAddress
 }
 
-// IsAllocated returns true if the server is currently allocated.
-func (gs *GameServer) IsAllocated() bool {
-	return gs.ObjectMeta.DeletionTimestamp == nil && gs.Status.State == GameServerStateAllocated
+// IsDeletable returns false if the server is currently allocated/reserved and should not be deleted
+func (gs *GameServer) IsDeletable() bool {
+	return !(gs.ObjectMeta.DeletionTimestamp.IsZero() &&
+		(gs.Status.State == GameServerStateAllocated || gs.Status.State == GameServerStateReserved))
 }
 
 // IsBeingDeleted returns true if the server is in the process of being deleted.
 func (gs *GameServer) IsBeingDeleted() bool {
 	return !gs.ObjectMeta.DeletionTimestamp.IsZero() || gs.Status.State == GameServerStateShutdown
 }
 
+// IsAllocatable is this GameServer in a state that allows it to be allocated?
+func (gs *GameServer) IsAllocatable() bool {
```

Comment:
CanBeAllocated perhaps?

Reply:
That reads much better :+1: 

`if gs.CanBeAllocated() { ... }`

Reply:
Actually,I just realised this never gets used _anywhere_ :man_facepalming:  so I'm going to delete it.